vinegar: added segregrevert, mouselock patch for wine 9.2; vinegar 1.6.1 -> 1.7.3

ardishko opened the pull request:
## Description of changes

<!--
Added the segregrevert patch for the wine installation vinegar ships with which is essential for roblox to run at all under wine.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

ardishko committed:
vinegar: add regregrevert patch for wine 9.0 (stable)

ardishco-the-great commented:
Oops, the PR is targeting unstable. My bad. I'll change the target and add the patch file in a second.

ardishko committed:
vinegar: added segregrevert.mypatch

SystematicError commented on pkgs/by-name/vi/vinegar/package.nix:
```diff
@@ -8,22 +8,16 @@
 let
-  # wine-staging doesn't support overrideAttrs for now
-  wine = wineWowPackages.stagingFull.overrideDerivation (oldAttrs: {
+  wine = wine64Packages.stable.overrideDerivation (oldAttrs: {
```
Why is wine64 being used instead of wineWow? wineWow provides 32 and 64 bit support whilst wine64 only provides support for 64 bit. Certain bits of Roblox still require 32 bit.

Also, the [vinegar wiki](https://vinegarhq.org/Home/faq.html#what-is-the-best-configuration) recommends using wine staging due to certain patches and features it provides.

ardishco-the-great replied:
Okay, I've switched the wine version to staging since it has the required fixes and the patches from the flatpak can properly apply.

kirillrdy replied:
is there a resolved clarification about switch to `wine64` ?

liquidovski replied:
Vinegar now requires the ``wine64`` unlike before which wine64 now provides

ardishko committed:
vinegar: switched Wine version to wine64Packages.staging, added
segregrevert patch, added mouselock patch, added
wine-loader-prefer-winedllpath.patch, bumped vinegar version from 1.6.1
--> 1.7.3 and removed patches folder

ardishko committed:
vinegar: removed wine-loader-prefer-winedllpatch.patch

ardishco-the-great commented:
I've removed the wine-dll-prefer-winedll.patch as it was mentioned that this patch is a flatpak-specific patch only needed for the flatpak for vinegar thus not really being necessary. My testing also confirms that It's not necessary to any core functionality of this program.

ardishko committed:
vinegar: added overrideAttrs comment back (minor change)

nyabinary commented:
1.7.4 is out.

liquidovski commented:
> 1.7.4 is out.

Updating the PR ASAP, thanks for the heads up

nyabinary commented:
@kirillrdy I don't think vinegar got updated to 1.7.4 here.

kirillrdy commented:
my bad, thanks for pointer @nyabinary !
